### PR TITLE
Add mul(x, sigmoid(x)) to swish substitution.

### PR DESCRIPTION
**Pull request by elad-c**
## Pull Request Description:
This PR adds a substitution for replacing mul(x, sigmoid(x)) with swish.

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/sigmoid_mul_to_swish.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    Substitutes mul(x, sigmoid(x)) with swish.
+    """
+
+    def __init__(self):
+        """
+        Initializes the MulSigmoidToSwish substitution matcher instance.
+        """
+        mul_matcher = NodeOperationMatcher(tf.math.multiply) | NodeOperationMatcher(Multiply)
+        sigmoid_matcher = NodeOperationMatcher(tf.sigmoid)
```
Comment (reuvenperetz) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/sigmoid_mul_to_swish.py:44]: What about the Activation layer with the string 'sigmoid'?

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/sigmoid_mul_to_swish.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        graph.add_node(swish_node)
+
+        # Replace functional conv node (and potentially add node) with Conv node.
+        graph.reconnect_in_edges(sigmoid_node, swish_node)
+        graph.reconnect_out_edges(mul_node, swish_node)
+        graph.replace_output_node(current_node=mul_node, new_node=swish_node)
+        graph.remove_edge(input_node, mul_node)
+        graph.remove_edge(sigmoid_node, mul_node)
+        graph.remove_node(sigmoid_node)
+        graph.remove_node(mul_node)
```
Comment (reuvenperetz) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/sigmoid_mul_to_swish.py:80]: If sigmoid is an output for the model it will fail...

File: tests/keras_tests/models_tests/test_networks_runner_float.py [hunk 3]
```diff
@@ -92,70 +92,70 @@ def run_network(self, model_float, input_shapes):
...
-        input_shapes = [[32, 224, 224, 3]]
-        from tensorflow.keras.applications.mobilenet import MobileNet
-        self.run_network(MobileNet(), input_shapes)
-
-    def test_mobilenet_v1_gptq(self):
-        input_shapes = [[16, 224, 224, 3]]
-        from tensorflow.keras.applications.mobilenet import MobileNet
-        self.run_network(MobileNet(), input_shapes)
+    # def test_mobilenet_v1(self):
+    #     input_shapes = [[32, 224, 224, 3]]
```
Comment (reuvenperetz) [inline tests/keras_tests/models_tests/test_networks_runner_float.py:96]: Why this?

File: tests/keras_tests/models_tests/test_networks_runner.py [hunk 4]
```diff
@@ -192,41 +192,41 @@ def test_resnet(self):
...
+    #     input_shapes = [[10, 224, 224, 3]]
+    #     num_calibration_iter = 1
+    #     from tensorflow.keras.applications.resnet_v2 import ResNet50V2
+    #     self.run_network(ResNet50V2(), input_shapes, num_calibration_iter)
+    #
+    # def test_densenet121(self):
+    #     input_shapes = [[10, 224, 224, 3]]
+    #     num_calibration_iter = 1
+    #     from tensorflow.keras.applications.densenet import DenseNet121
+    #     self.run_network(DenseNet121(), input_shapes, num_calibration_iter)
```
Comment (reuvenperetz) [inline tests/keras_tests/models_tests/test_networks_runner.py:211]: Why this?

**Review by reuvenperetz — COMMENTED** [review]
We do not need such a substituion for torch?

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/sigmoid_mul_to_swish.py (hunk 2 above)
Comment (elad-c) [reply model_compression_toolkit/core/keras/graph_substitutions/substitutions/sigmoid_mul_to_swish.py:80]: Might be relevant for many substitutions... @ofirgo

**Review by reuvenperetz — APPROVED** [review]
